src/main/java/org/concordion/internal/command/VerifyRowsCommand.java

```diff
@@ -49,42 +52,98 @@ public void verify(CommandCall commandCall, Evaluator evaluator, ResultRecorder
+        Class<? extends VerificationStrategy> clazz = detectStrategyClass(commandCall);
+        try {
+            clazz.getConstructor(
+                    CommandCall.class,
+                    Evaluator.class,
+                    ResultRecorder.class,
+                    Announcer.class,
+                    String.class,
+                    Iterable.class
+            ).newInstance(commandCall, evaluator, resultRecorder, listeners, loopVariableName, iterable).verify();
+        } catch (Exception e) {
+            throw new RuntimeException("Verification strategy must declare constructor with arguments: " +
+                    "CommandCall commandCall, Evaluator evaluator, ResultRecorder resultRecorder,\n" +
+                    "Announcer<VerifyRowsListener> listeners, String loopVariableName, Iterable<Object> actualRows");
         }
-        
-        for (; index < detailRows.length; index++) {
-            Row detailRow = detailRows[index];
-            resultRecorder.record(Result.FAILURE);
-            announceMissingRow(detailRow.getElement());
+    }
+
+    private static final String DEFAULT_STRATEGIES_PACKAGE = "org.concordion.internal.command.strategies.";
+
+    private Class<? extends VerificationStrategy> detectStrategyClass(CommandCall commandCall) {
+        String strategy = findFirstExistingAttributeValue(commandCall, "verificationStrategy", "verification-strategy");
+        if (strategy == null) {
+            return Default.class;
         }
+        return findFirstExistingClassOrDefault(DEFAULT_STRATEGIES_PACKAGE + strategy, strategy);
     }
-    
-    private void announceExpressionEvaluated(Element element) {
-        listeners.announce().expressionEvaluated(new ExpressionEvaluatedEvent(element));
+
+    private String findFirstExistingAttributeValue(CommandCall commandCall, String... attributeNames) {
+        for (String attributeName : attributeNames) {
+            String value = commandCall.getElement().getAttributeValue(attributeName, ConcordionBuilder.NAMESPACE_CONCORDION_2007);
+            if (value != null) {
+                return value;
+            }
+        }
+        return null;
     }
 
-    private void announceMissingRow(Element element) {
-        listeners.announce().missingRow(new MissingRowEvent(element));
+    private Class<? extends VerificationStrategy> findFirstExistingClassOrDefault(String... names) {
+        for (String name : names) {
+            try {
+                Class<?> aClass = Class.forName(name);
+                if (VerificationStrategy.class.isAssignableFrom(aClass)) {
+                    return (Class<? extends VerificationStrategy>) aClass;
+                }
+            } catch (ClassNotFoundException e) {}
+        }
+        return Default.class;
     }
 
-    private void announceSurplusRow(Element element) {
-        listeners.announce().surplusRow(new SurplusRowEvent(element));
+    public static abstract class VerificationStrategy {

```

Comment:
Please make this a top level class. Since it's an extension point, I think we should also move it to the api package.